blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/MultiBranchPipelineImpl.java

```diff
@@ -133,24 +133,39 @@ public Integer getWeatherScore(){
         /**
          * TODO: this code need cleanup once MultiBranchProject exposes default branch. At present
          *
-         * At present we look for master as primary branch, if not found we find the latest build and return
-         * its score.
+         * At present we look for master as primary branch, if not found we find the latest build across all branches and
+         * return its score.
          *
          * If there are no builds taken place 0 score is returned.
          */
 
-        Job j = mbp.getBranch("master");
+        Job j = mbp.getItem("master");
         if(j == null) {
-            j = mbp.getBranch("production");
-            if(j == null){ //get latest
+            j = mbp.getItem("production");
```

Comment:
'production' branch. if not master then production, heuristic we came up with in absence of default branch concept available in underlying model.